dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNIIDMExtensions.hpp

```diff
@@ -66,23 +78,66 @@ struct IIDMExtensions {
    */
   template<class T>
   static ExtensionDefinition<T> getExtension(const std::string& libPath) {
-    boost::dll::shared_library extensionLibrary(libPath);
+    std::shared_ptr<boost::dll::shared_library> extensionLibrary;
+    std::unique_lock<std::mutex> lock(librariesMutex_);
+    if (libraries_.count(libPath) > 0) {
+      extensionLibrary = libraries_.at(libPath);
+    } else {
+      try {
+        extensionLibrary = std::make_shared<boost::dll::shared_library>(libPath);
+      } catch (const std::exception&) {
+        // No log here as if extension is not defined, it is not defined for all elements and this function is called for
+        // every network element supporting an external extension
+        return buildDefaultExtensionDefinition<T>();
+      }
+      libraries_[libPath] = extensionLibrary;
+    }
+    lock.unlock();
     const auto& name = IIDMExtensionTrait<T>::name;
 
     std::string createName = "create" + std::string(name);
-    if (!extensionLibrary.has(createName)) {
-      throw DYNError(DYN::Error::GENERAL, LibraryLoadFailure, libPath + "::" + createName);
+    if (!extensionLibrary->has(createName)) {
+      // warning here because if the library can be loaded, all extensions should be defined in it
```

Comment:
this does not explain why it is a warning and not an error

Reply:
It is a warning because if the extension is not implemented, it may not be a problem for the simulation